### DISCUSSION TITLE
Migrating or copying libraries from one pypiserver to another

### DISCUSSION THREAD

**Discussion by jfacevedo** [pypiserver/pypiserver#606] [General]
There is a slightly old pypiserver that has been in use for some time and already hosts quite a few libraries.

I have decided to deploy a new instance with the latest version of pypiserver and a more updated AMI.

My question now is, what are the ways to migrate or copy the libraries hosted by the current pypiserver to the new one?

scp could be an option? or is there a tool that downloads the libraries from the current instance and uploads them to the new one?

**Comment by dee-me-tree-or-love** [comment]
Hello @jfacevedo! I'm very sorry for taking a long time to get to this. This is a great question! 

Seeing that you have closed the discussion, have you managed to find a solution that works for you? 
I would be very curious to read it. 

In general, I don't think there would be large conflicts as to how the packages are stored since I can't recall changing much of the file-system operations of `pypi-server`. So as long as the file arrangement is kept in-sync it should work smoothly after porting. 

The options on how to migrate/duplicate the files from one location to the other would depend on the deployment details. I believe `scp` should work very well for the direct relocation, e.g. between two server instances. If you are using some more advanced/cloud-y volume technologies, I presume some cloud providers should have specialised/out-of-the-box ways to do this too. 

I'm very curious what did you end up doing in your use case if you are willing to share 😊
